R/checks.R

```diff
@@ -598,3 +612,30 @@ check_no_tuning <- function(x) {
   rlang::abort(msg, call = NULL)
 }
 
+
+check_eval_times <- function(.times, metrics) {
+  metric_types <- tibble::as_tibble(metrics)$class
+  has_dyn <- any(metric_types == "dynamic_survival_metric")
+  if (!is.null(.times) & !has_dyn) {
+    rlang::abort(
+      "Evaluation times are only used for dynamic survival metrics.",
+      call = NULL
+    )
+  }
+  if (is.null(.times) & has_dyn) {
+    rlang::abort(
+      "One or more metric requires the specification of time points in the `eval_times` argument.",
+      call = NULL
+    )
+  }
+
+  # will still propagate nulls
+  .times <- .times[!is.na(.times)]
+  .times <- unique(.times)
+  .times <- sort(.times)
+  .times <- .times[.times >= 0]
+  if (identical(.times, numeric(0))) {
+    rlang::abort("There were no usable evaluation times.", call = NULL)
```

Comment:
Could this message provide some guidance (or link to docs) on what makes a usable evaluation time?

Reply:
This looks like we might want to also avoid `Inf`